android/quest/src/main/java/org/smartregister/fhircore/quest/configuration/parser/G6PDDetailConfigParser.kt

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2021 Ona Systems, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.smartregister.fhircore.quest.configuration.parser
+
+import android.content.Context
+import android.content.Intent
+import ca.uhn.fhir.rest.gclient.TokenClientParam
+import com.google.android.fhir.FhirEngine
+import com.google.android.fhir.logicalId
+import com.google.android.fhir.search.search
+import org.hl7.fhir.r4.model.CodeableConcept
+import org.hl7.fhir.r4.model.Coding
+import org.hl7.fhir.r4.model.Condition
+import org.hl7.fhir.r4.model.Observation
+import org.hl7.fhir.r4.model.Questionnaire
+import org.hl7.fhir.r4.model.QuestionnaireResponse
+import org.hl7.fhir.r4.model.ResourceType
+import org.smartregister.fhircore.engine.util.extension.asDdMmmYyyy
+import org.smartregister.fhircore.engine.util.extension.valueToString
+import org.smartregister.fhircore.quest.configuration.view.PatientDetailsViewConfiguration
+import org.smartregister.fhircore.quest.configuration.view.Properties
+import org.smartregister.fhircore.quest.configuration.view.Property
+import org.smartregister.fhircore.quest.data.patient.model.AdditionalData
+import org.smartregister.fhircore.quest.data.patient.model.QuestResultItem
+import org.smartregister.fhircore.quest.ui.patient.details.SimpleDetailsActivity
+
+class G6PDDetailConfigParser(fhirEngine: FhirEngine) : DetailConfigParser(fhirEngine) {
```

Comment:
can we use something generic here? having a g6pd specific parser doesn't align with the goals of making this reusable and not require coding

CC @ellykits @maimoonak

Reply:
@shoaibmushtaq25 @owais-vd we can use or extend this (may be some modifications would be required to allow handling both). This way we would also get rid of hardcoded strings below

Reply:
@owais-vd Yes, I am on it.

Reply:
@pld For your context, As per today's discussion with @f-odhiambo @owais-vd , @maimoonak and @kwasim , we are planning to merge this PR with the G6PD Parser implementation in order to do a release for it and for the generic solution, we will be picking it as a separate issue, The work on generic solution is already under progress on my local branch.

Reply:
understood, as discussed on main thread the related issue is here, https://github.com/opensrp/fhircore/issues/961

Reply:
I will not review the code in this file